integration_test/gce/gce_testing.go

```diff
@@ -802,7 +811,32 @@ func RunRemotely(ctx context.Context, logger *log.Logger, vm *VM, stdin string,
 	args = append(args, "-oIdentityFile="+privateKeyFile)
 	args = append(args, sshOptions...)
 	args = append(args, wrappedCommand)
-	return runCommand(ctx, logger, stdin, args)
+	return startCommand(ctx, logger, stdin, args, wait)
+}
+
+// RunRemotely runs a command on the provided VM.
+// The command should be a shell command if the VM is Linux, or powershell if the VM is Windows.
+// Returns the combined stdout+stderr as a string, plus an error if there was
+// a problem.
+//
+// 'command' is what to run on the machine. Example: "cat /tmp/foo; echo hello"
+// 'stdin' is what to supply to the command on stdin. It is usually "".
+// TODO: Remove the stdin parameter, because it is hardly used.
+func RunRemotely(ctx context.Context, logger *log.Logger, vm *VM, stdin string, command string) (_ CommandOutput, err error) {
+	return runRemotely(ctx, logger, vm, stdin, command, true)
+}
+
+// StartRemotely starts a command on the provided VM.
+// The command should be a shell command if the VM is Linux, or powershell if the VM is Windows.
+// Returns the combined stdout+stderr as a string, plus an error if there was
+// a problem.
+// If the wait argument is true, the function waits for the command to finish running. Otherwise,
+// it returns immediately.
+// 'command' is what to run on the machine. Example: "cat /tmp/foo; echo hello"
+// 'stdin' is what to supply to the command on stdin. It is usually "".
+// TODO: Remove the stdin parameter, because it is hardly used.
+func StartRemotely(ctx context.Context, logger *log.Logger, vm *VM, stdin string, command string, wait bool) (_ CommandOutput, err error) {
```

Comment:
Can we avoid the `StartRemotely` and `startCommand` stuff by just following what the Linux version of the fluentd-forward test does?

Reply:
Fluent Bit won't read from stdin on Windows, meaning we have to run it as a background process. @shafinsiddique that means `cmd.Run` blocks forever because Fluent Bit keeps running right?

Reply:
Even if it doesn't read from stdin, there are simpler workarounds available, e.g. write a single line to a text file and tail it.

Reply:
That doesn't solve the problem I don't think. The reason reading from stdin worked before if I understand correctly is that it ran as a one off and then concluded when `stdin` was done. Whether it's `random` input or `tail` it can't just do the same one-off operation it does on Linux.

Reply:
> Fluent Bit won't read from stdin on Windows, meaning we have to run it as a background process. @shafinsiddique that means `cmd.Run` blocks forever because Fluent Bit keeps running right?

Yes, cmd.Run is blocking

Reply:
> Starting the process is already a one-off operation. It's not clear why `RunRemotely` isn't sufficient here...
> 
> `&` probably won't work since I think processes started that way are bound to the shell, but certainly the WMI way should work to keep it backgrounded (`Invoke-WmiMethod -Class Win32_Process -Name Create -ArgumentList 'fluent-bit.exe' ...`)

We CAN use RunRemotely if we change it's implementation to call cmd.Start() instead of cmd.Run(). Is that what you're suggesting?

Reply:
> `&` probably won't work since I think processes started that way are bound to the shell, but certainly the WMI way should work to keep it backgrounded (`Invoke-WmiMethod -Class Win32_Process -Name Create -ArgumentList 'fluent-bit.exe' ...`)

OH i see what you mean, okay this is a good idea